### PR TITLE
Created entities and deleted .gitignore from folders

### PR DESCRIPTION
**Pull request by marceldobrica**

### CODE REVIEW

File: src/Entity/Programme.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    private \DateTime $startDate;
+
+    private \DateTime $endDate;
+
+    private ?User $trainer;
+
+    private Room $room;
+
+    private ?Collection $customers;
```
Comment (Raresmldvn) [inline src/Entity/Programme.php]: why is this nullable?

File: src/Entity/Programme.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+        return $this;
+    }
+
+    public function getTrainer(): ?User
+    {
+        return $this->trainer;
+    }
+
+    public function setTrainer(User $trainer): self
```
Comment (Raresmldvn) [inline src/Entity/Programme.php]: `?User $trainer`. if the property can be null, you must allow the setter to set it to null

File: src/Entity/User.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+class User
+{
+    private int $id;
+
+    private string $password;
+
+    private Collection $roles;
+
+    private string $cnp;
```
Comment (Raresmldvn) [inline src/Entity/User.php]: why is this private? I don't see any reason to keep it private

File: src/Entity/User.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    public string $lastName = '';
+
+    private Collection $programs;
+
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    public function setId(int $id): void
```
Comment (Raresmldvn) [inline src/Entity/User.php]: the id should never be changed

File: src/Entity/User.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+    private Collection $roles;
+
+    private string $cnp;
+
+    public string $firstName = '';
+
+    public string $lastName = '';
+
+    private Collection $programs;
```
Comment (Raresmldvn) [inline src/Entity/User.php]: if the entity is called `Programme`, you should be consistent and call this property `programmes`

File: src/Entity/User.php [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        $this->id = $id;
+    }
+
+    public function getPassword(): string
+    {
+        return $this->password;
+    }
+
+    public function setPassword(string $password): void
```
Comment (Raresmldvn) [inline src/Entity/User.php]: why don't you return `$this` in the case of this method?

File: src/Entity/Programme.php (hunk 1 above)
Comment (marceldobrica) [reply src/Entity/Programme.php]: A program may have no customers in a specific moment (when created for example).
Also a program may have no trainer when it is imported... The trainer will be set up by management.
Programs without trainers should not be visible for apoinments.

File: src/Entity/Programme.php [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+    private \DateTime $endDate;
+
+    private ?User $trainer;
+
+    private Room $room;
+
+    private ?Collection $customers;
+
+    public bool $isOnline;
```
Comment (Raresmldvn) [inline src/Entity/Programme.php]: initialize this one too

**Review by Raresmldvn — CHANGES_REQUESTED** [review]

File: src/Entity/Programme.php (hunk 1 above)
Comment (Raresmldvn) [reply src/Entity/Programme.php]: ok. but a programme with no customers means a program with an empty collection. An empty collection means a collection with no element, not a null object (think about the difference between an empty array and a null) :smiley:

File: src/Entity/Programme.php (hunk 7 above)
Comment (marceldobrica) [reply src/Entity/Programme.php]: Ok

File: src/Entity/User.php (hunk 6 above)
Comment (marceldobrica) [reply src/Entity/User.php]: Ok. I have used Code->generate setters and getters and then refactor...

File: src/Entity/User.php (hunk 5 above)
Comment (marceldobrica) [reply src/Entity/User.php]: OK

File: src/Entity/User.php (hunk 3 above)
Comment (marceldobrica) [reply src/Entity/User.php]: Not a good reason... gdpr ...

File: src/Entity/Programme.php (hunk 2 above)
Comment (marceldobrica) [reply src/Entity/Programme.php]: OK.

File: src/Entity/Room.php [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+class Room
+{
+    private int $id;
+
+    public string $name;
+
+    public int $capacity;
+
+    public Building $build;
```
Comment (rusdavid1) [inline src/Entity/Room.php]: I think this should be private

File: src/Entity/Programme.php [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+use Doctrine\Common\Collections\Collection;
+
+class Programme
+{
+    private int $id;
+
+    public string $name = '';
+
+    private \DateTime $startDate;
```
Comment (rusdavid1) [inline src/Entity/Programme.php:15]: You should probably have consistent naming on the DateTime properties. For example, in building you have startTime and here startDate. (I had the same issue myself :) )

Comment (marceldobrica) [reply src/Entity/Programme.php:15]: I follow our diagram.... I think it is ok.... no such a problem ... but you are right...

**Pull request merged**